azurelinuxagent/common/osutil/default.py

```diff
@@ -516,8 +516,15 @@ def set_dhcp_hostname(self, hostname):
                                       'send host-name',
                                       'send host-name "{0}";'.format(hostname))
 
-    def restart_if(self, ifname):
-        shellutil.run("ifdown {0} && ifup {1}".format(ifname, ifname))
+    def restart_if(self, ifname, retries=3, wait=5):
+        for attempt in range(1, retries+1, 1):
+            return_code=shellutil.run("ifdown {0} && ifup {1}".format(ifname, ifname))
```

Comment:
`return_code=shellutil.run("ifdown {0} && ifup {0}".format(ifname))` would be clearer.